### ISSUE TITLE
Config option idea for vehicles

### ISSUE THREAD

**Issue by pixelrider2000** [Unknown025/Flans-Mod-Plus#64]
I think it would be cool if you could make a vehicle invulnerable to fist damage/damage by hand (e.g. with a config option). The only way to break it still could be with one of flan's debug sticks or something like that. Sometimes I think it breaks the atmosphere when a vehicle is being broken by hand. 

Let me know what you think :)

**Comment by GoldSloth** [comment]
It's a good idea, not sure how to go about handling the item that can break it though. Should it be configurable, if so how? Or hardcoded to an item that's added in, if so what item? Like, "keys"?

**Comment by pixelrider2000** [comment]
I think a hardcoded item would work pretty well in this case (e.g. the "Stick of Destruction" that you get when typing the command "/teams getSticks"). I'm not sure about configurable item but I let you now once I have something.

**Comment by pixelrider2000** [comment]
I now have an idea for configurable items. Perhaps there could be a new config option (e.g. canBreakVehicles) for tools. And when a player in creative mode and with op permissions clicks on a vehicle with this item, it will link the item to the vehicle. Perhaps you could even make it so when the vehicle is shift right clicked the item will not only be linked to this one specific, but to all vehicles of the same type. What I mean by that is when you shift right click a "Tiger I" tank, the item would work on all "Tiger I" tanks and not only on that specific one. 

Just an idea, let me know what you think ;)

**Comment by GoldSloth** [comment]
I like the idea of tools being kinda 'keys' to destroy but maybe even enter. You could maybe specify a list of shortnames that it works with. Going into individual vehicles or classes of vehicles gets more complicated though, because it'd need state. That's where a proper key/locking system comes in. What would you find helpful, from those?

**Comment by pixelrider2000** [comment]
I think keys are a good idea too, but I would have this feature seperate from the other, so a "fist-proof" vehicle doesn't have to have a key and vice versa. Do you mean a list of names of vehicles in the config file of the tool by "list of shortnames"? If that's the case you are probably saying that the tool would work on all vehicles, that have their name in the shortname list? For the breaking feature I don't think it'd be too big of a deal if individual vehicles weren't supported, but I'm not too sure on keys though. The reason to have a key in the first place would be so that you are the only person who has access to your vehicle right?

Please tell me if I misunderstood you though.

**Comment by GoldSloth** [comment]
Yeah, that's what I mean by list of shortnames. There could also be an option to break all. For now we should probably stick with an item that can break all, because only breaking some sounds a bit odd to me.

**Comment by pixelrider2000** [comment]
I agree, that sounds good to me. One question though, what do you think of a hardcoded "master tool" that can break all vehicles (Stick of Destruction)?
